docs/markdown/Using Pants/environments.md

```diff
@@ -0,0 +1,221 @@
+---
+title: "Environments: Cross-Platform or Remote Builds"
+slug: "environments"
+hidden: false
+createdAt: "2022-10-03T21:39:51.235Z"
+updatedAt: "2022-10-03T21:39:51.235Z"
+---
+Environments
+============
+
+By default, Pants will execute all sandboxed build work directly on localhost. But defining and using additional "environments" for particular targets allows Pants to transparently execute some or all of your build either:
+1. locally in Docker containers
+2. remotely via [remote execution](doc:remote-execution)
+3. locally, but with a non-default set of environment variables and settings (such as for cross-building)
+
+## Defining environments
+
+Environments are defined using environment targets:
+
+* [`local_environment`](doc:reference-local_environment) - Runs without containerization on localhost (which is also the default if no environment targets are defined).
+* [`docker_environment`](doc:reference-docker_environment) - Runs in a cached container using the specified Docker image.
+* [`remote_environment`](doc:reference-remote_environment) - Runs in a remote worker via [remote execution](doc:remote-execution) (possibly with containerization, depending on the server implementation).
+
+Environment targets are given short, descriptive names using the [`[environments.names]` option](doc:reference-environments#names) (usually defined in `pants.toml`), which consuming targets use to refer to them in `BUILD` files. That might look like a `pants.toml` section and `BUILD` file (at the root of the repository in this case) containing:
+
+```toml
+[environments.names]
+linux = "//:local_linux"
+linux_docker = "//:local_busybox"
+```
+
+```python
+local_environment(
+  name="local_linux",
+  compatible_platforms=["linux_x86_64"],
+  fallback_environment="local_busybox",
+  ..
+)
+
+docker_environment(
+  name="local_busybox",
+  platform="linux_x86_64",
+  image="busybox:latest@sha256-abcd123...",
+  ..
+)
+```
+
+### Environment-aware options
+
+Environment targets have fields (target arguments) which correspond to options which are marked "environment-aware". When an option is environment-aware, the value of the option that will be used in an environment can be overridden by setting the corresponding field value on the environment target for that environment. If an environment target does not set a value, it defaults to the value which is set globally via options values.
+
+For example, the [`[python-bootstrap].search_path` option](doc:reference-python-bootstrap#search_path) is environment-aware, which is indicated in its help. It can be overridden for a particular environment by a corresponding environment target field, such as [the one on `local_environment`](doc:reference-local_environment#codepython_bootstrap_search_pathcode).
```

Comment:
It would probably help to have a `BUILD` file example where you set this, and then have a comment in it saying "this overrides the option `[python-bootstrap].search_path`

People tend to really benefit from examples